src/planner_test.cc

```diff
@@ -194,9 +204,42 @@ class PlannerHarness {
   Planner *planner_;
 };
 
+// Verify that on average, the profile satisfies
+// the configured constraints both for speed and acceleration.
+static void VerifySegmentConstraints(const MachineControlConfig &config,
+                                     const LinearSegmentSteps &segment,
+                                     const float rel_error = 1e-3) {
+  // The axes X, Y, Z are mapped to motors 1, 2, 3.
+  for (int i = 1; i <= AXIS_Z; ++i) {
+    const GCodeParserAxis axis = (GCodeParserAxis)i;
+    if (segment.steps[axis]) {
+      const double accel =
+        fabs_accel(segment.v0, segment.v1, segment.steps[axis]);
+      const double max_accel =
+        config.acceleration[axis] * config.steps_per_mm[axis];
+      const double max_feedrate =
+        config.max_feedrate[axis] * config.steps_per_mm[axis];
+      EXPECT_LE(segment.v0, max_feedrate);
+      EXPECT_LE(segment.v1, max_feedrate);
+      if (segment.v0 != segment.v1)
+        EXPECT_NEAR_REL(accel, max_accel, rel_error);
```

Comment:
Maybe let's test with `EXPECT_LE`?

Reply:
Yes, as we only want to know that we never exceed the acceleration, but smaller is ok (in particular as we'll have jerk handling at some point, in which we expect it to be slower initially)

Reply:
(solved)